kafka-event-listener/src/main/java/org/hypertrace/core/kafka/event/listener/KafkaEventListenerThread.java

```diff
@@ -0,0 +1,133 @@
+package org.hypertrace.core.kafka.event.listener;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import com.typesafe.config.ConfigValue;
+import io.micrometer.core.instrument.Counter;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Properties;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.PartitionInfo;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.errors.InterruptException;
+import org.apache.kafka.common.serialization.ByteArrayDeserializer;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.hypertrace.core.serviceframework.metrics.PlatformMetricsRegistry;
+
+@Slf4j
+class KafkaEventListenerThread<K, V> extends Thread {
+  private static final String EVENT_CONSUMER_ERROR_COUNT = "event.consumer.error.count";
+  private static final String TOPIC_NAME = "topic.name";
```

Comment:
Out of curiosity, does kafka define these anywhere?

Reply:
General kafka configs are present under for consumer at `org.apache.kafka.clients.consumer.ConsumerConfig`. For kafka streams they are at `org.apache.kafka.streams.StreamsConfig`

Reply:
I did not find an alias for topic name in the CommonClientConfigs or the above two classes. @laxmanchekka do we have one?

Reply:
No built-in kafka consumer configs for topic. It's explicitly subscribed in application code via subscribe api.